pkg/server/commands/list_objects.go

```diff
@@ -306,6 +310,9 @@ func (q *ListObjectsQuery) evaluate(
 				Consistency:      req.GetConsistency(),
 			}, reverseExpandResultsChan, reverseExpandResolutionMetadata)
 			if err != nil {
+				if errors.Is(err, context.DeadlineExceeded) {
+					resolutionMetadata.DidTimeOut = true
```

Comment:
Since this is inside go func, do we need to protect DidTimeout via mutex / atomic?   (i.e., concern whether in some cases go panic due to write after write when we have detect race turned on). This was the reason why we have `WasThrottled` being `atomic.Bool`.

Reply:
Unit test to prove it is to have multi goroutine running evaluate have deadline exceeded